src/commands/synthetics/mobile.ts

```diff
@@ -83,24 +83,28 @@ export const uploadApplicationAndOverrideConfig = async (
   overriddenTestsToTrigger: TestPayload,
   uploadedApplicationByPath: {[applicationFilePath: string]: {applicationId: string; fileName: string}[]}
 ): Promise<void> => {
-  if (!userConfigOverride.mobileApplicationVersionFilePath) {
+  if (!(userConfigOverride.mobileApplicationVersionFilePath || userConfigOverride.mobileApplicationVersion)) {
```

Comment:
very small nit, but I would avoid doing the `not(A || B)`, and do `not A && not B` instead, as it feels easier to  understand!

Reply:
Actually this condition is not needed, I will remove it.